In [4]:
import nbimporter

In [5]:
import numpy as np

In [6]:
from math import *

In [7]:
from Formula import *

#### Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [8]:
vyb = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

####  среднее арифметическое

In [9]:
ma = sum(vyb)/len(vyb)
ma

65.3

Ответ: 65.3

#### среднее квадратичное отклонение (несмещенное, т.к. объем < 100)

In [10]:
np.sqrt(sum([(i-ma)**2 for i in vyb])/(len(vyb)-1))
    

31.624607341019814

Ответ: 31.62

#### смещенная оценка дисперсии

In [11]:
sum([(i-ma)**2 for i in vyb])/len(vyb)

950.11

Ответ: 950.11

#### несмещенная оценка дисперсии

In [12]:
sum([(i-ma)**2 for i in vyb])/(len(vyb)-1)

1000.1157894736842

Ответ:1000.11

#### В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

- 2 мяча из первой корзины и 1 из второй: $ \frac{С^{2}_{5}}{С^{2}_{8}} * \frac{С^{1}_{5}С^{3}_{7}}{C^{4}_{12}}$<br><br>
- 1 мяч из первой корзины и 2 мяча из второй: $ \frac{С^{1}_{5}С^{1}_{3}}{C^{2}_{8}} * \frac{С^{2}_{5}С^{2}_{7}}{C^{4}_{12}} $<br><br>
- 3 мяча из третьей корзины: $ \frac{С^{2}_{3}}{С^{2}_{8}} * \frac{С^{3}_{5}С^{1}_{7}}{C^{4}_{12}} $<br><br>
$ \frac{С^{2}_{5}}{С^{2}_{8}} * \frac{С^{1}_{5}С^{3}_{7}}{C^{4}_{12}} + \frac{С^{1}_{5}С^{1}_{3}}{C^{2}_{8}} * \frac{С^{2}_{5}С^{2}_{7}}{C^{4}_{12}} + \frac{С^{2}_{3}}{С^{2}_{8}} * \frac{С^{3}_{5}С^{1}_{7}}{C^{4}_{12}} $

In [13]:
((soch(2, 5)/soch(2, 8))*((soch(1, 5)*soch(3, 7)/soch(4, 12)))+((soch(1, 5)*soch(1, 3)/soch(2, 8)))*\
((soch(2, 5)*soch(2, 7)/soch(4, 12)))+(soch(2, 3)/soch(2, 8))*((soch(3, 5)*soch(1, 7)/soch(4, 12))))*100

36.868686868686865

Ответ: 36.86%

#### На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

Для данной задачи применима формула Байеса, т.к. попадание уже произошло<br>
$ P(B|A)=\frac{P(A|B)*P(B)}{P(A)} $<br>
$P(B)$ = 0.333 для всех стрелков<br>
$P(A|B)$ = 0.9, 0.8, 0.6 для 1, 2, 3 стрелка соответственно<br>
$P(A)$ = 0.333 * 0.9 + 0.333 * 0.8 + 0.333 * 0.6

In [14]:
((0.9*0.333)/(0.333*0.9+0.333*0.8+0.333*0.6))*100

39.130434782608695

In [15]:
((0.8*0.333)/(0.333*0.9+0.333*0.8+0.333*0.6))*100

34.78260869565218

In [16]:
((0.6*0.333)/(0.333*0.9+0.333*0.8+0.333*0.6))*100

26.08695652173913

Ответ: 39.13%,  34.78%,  26.08%

#### В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

Для данной задачи применима формула Байеса, т.к. экзамен уже сдан.<br>
$ P(B|A)=\frac{P(A|B)*P(B)}{P(A)} $<br>
$P(B)$ = 0.25, 0.25, 0.5 для A, B, C соответственно<br>
$P(A|B)$ = 0.8, 0.7, 0.9 для A, B, C соответственно<br>
$P(A)$ = 0.25 * 0.8 + 0.25 * 0.7 + 0.5 * 0.9

In [17]:
((0.8*0.25)/(0.8*0.25+0.7*0.25+0.9*0.5))*100

24.242424242424246

In [18]:
((0.7*0.25)/(0.8*0.25+0.7*0.25+0.9*0.5))*100

21.21212121212121

In [19]:
((0.9*0.5)/(0.8*0.25+0.7*0.25+0.9*0.5))*100

54.545454545454554

Ответ: 24.24%,  21.21%,  54.54%

#### Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

#### а)  все детали
События наступают одновременно:<br> $P(B_3)=P(A_1)*P(A_2)*P(A_3)$

In [20]:
(0.1*0.2*0.25)*100

0.5000000000000001

Ответ: 0.5%

#### б) только две детали<br>
Либо 1 и 2, 3 - рабочая, либо 2 и 3, 1 - рабочая, либо 1 и 3, 2 - рабочая<br>
$P(B_2)=P(A_1)*P(A_2)*(1-P(A_3))+P(A_2)*P(A_3)*(1-P(A_1))+P(A_1)*P(A_3)**(1-P(A_2))$

In [21]:
two = 0.1*0.2*(1-0.25)+0.2*0.25*(1-0.1)+0.1*0.25*(1-0.2)
two*100

8.000000000000002

Ответ: 8%

#### в) хотя бы одна деталь<br>
Вероятность, обратная вероятности того, что ни одна не выйдет из строя<br>
$P(B_1) = 1-(1-P(A_1))*(1-P(A_2))*(1-P(A_3))$

In [22]:
(1-(0.9*0.8*0.75))*100

46.0

Ответ: 46%

####  г) от одной до двух деталей<br>
Либо одна, либо две:<br>
$P(B_{1,2})=P(B_1)+P(B_2)$
$P(B_1)=P(A_1)**(1-P(A_2)*(1-P(A_3)+P(A_2)*(1-P(A_1)*(1-P(A_3)+P(A_3)*(1-P(A_1)*(1-P(A_3)$

In [29]:
one = 0.1*(1-0.2)*(1-0.25)+0.2*(1-0.1)*(1-0.25)+0.25*(1-0.1)*(1-0.2)
one

0.375

In [24]:
(one + two)*100

45.5

Ответ: 45.5%